### PR TITLE
feat(builtin): add Kotlin example

### PR DESCRIPTION
**Pull request by alexeagle**
This uses the Kotlin JS output to build a web application

### CODE REVIEW

**Comment by alexeagle** [comment]
Windows failure
https://storage.googleapis.com/bazel-untrusted-buildkite-artifacts/370acd4e-2c6d-4574-9eed-1adbe444d434/examples%5Cexamples_kotlin%5Cattempt_1.log

```
INFO: Call stack for the definition of repository 'maven' which is a coursier_fetch (rule definition at D:/b/w7fbjuxn/execroot/build_bazel_rules_nodejs/_tmp/cd373d66a74770dafff62116e395fc8c/_bazel_b/3smcru4j/external/rules_jvm_external/coursier.bzl:475:18):
 - D:/b/w7fbjuxn/execroot/build_bazel_rules_nodejs/_tmp/cd373d66a74770dafff62116e395fc8c/_bazel_b/3smcru4j/external/rules_jvm_external/defs.bzl:59:5
 - D:/temp/tmp-7892ckmfuca059qf/WORKSPACE:39:1
ERROR: An error occurred during the fetch of repository 'maven':
   Unable to run coursier: java.io.IOException: ERROR: src/main/native/windows/process.cc(184): CreateProcessW("D:\b\w7fbjuxn\execroot\build_bazel_rules_nodejs\_tmp\cd373d66a74770dafff62116e395fc8c\_bazel_b\3smcru4j\external\rules_jvm_external\third_party\coursier\coursier"): %1 is not a valid Win32 application.
Internal error thrown during build. Printing stack trace: java.util.UnknownFormatConversionException: Conversion = '1'
	at java.base/java.util.Formatter.checkText(Unknown Source)
	at java.base/java.util.Formatter.parse(Unknown Source)
```

fyi @jin

**Comment by jin** [comment]
If you use pinned artifacts with rules_jvm_external, you can skip the execution of Coursier to download artifacts. 

https://github.com/bazelbuild/rules_jvm_external#pinning-artifacts-and-integration-with-bazels-downloader

File: examples/kotlin/WORKSPACE [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    sha256 = "f04b1466a00a2845106801e0c5cec96841f49ea4e7d1df88dc8e4bf31523df74",
+    strip_prefix = "rules_jvm_external-2.7",
+    url = "https://github.com/bazelbuild/rules_jvm_external/archive/2.7.zip",
+)
+
+load("@rules_jvm_external//:defs.bzl", "maven_install")
+
+maven_install(
+    name = "maven",
+    artifacts = [
```
Comment (jin) [inline examples/kotlin/WORKSPACE]: Consider adding a comment with instructions to update maven_install.json

```
# Run 'bazel run @unpinned_maven//:pin` after updating this list to create a new maven_install.json
```

Running this command is not very intuitive, and can save people a few minutes from reading the docs.

**Comment by alexeagle** [comment]
@jin thanks!
next build failure https://storage.googleapis.com/bazel-untrusted-buildkite-artifacts/7f7ac926-7fc7-491a-b770-f75814b5e644/examples%5Cexamples_kotlin%5Cattempt_1.log
```
ERROR: D:/temp/tmp-64845eeqsguplvgk/BUILD.bazel:10:1: Action kotlinx-html-js.js failed (Exit 2)
--
  | LAUNCHER ERROR: Cannot launch process: "python.exe" D:\b\bpzbl574\execroot\build_bazel_rules_nodejs\_tmp\cd373d66a74770dafff62116e395fc8c\_bazel_b\v2ui4k2e\execroot\kotlin_example\bazel-out\host\bin\external\io_bazel_rules_kotlin\kotlin\internal\js\importer.zip --jar bazel-out/x64_windows-fastbuild/bin/external/maven/v1/http/dl.bintray.com/kotlin/kotlinx.html/org/jetbrains/kotlinx/kotlinx-html-js/0.6.12/kotlinx-html-js-0.6.12.jar --out bazel-out/x64_windows-fastbuild/bin/kotlinx-html-js.js --aux bazel-out/x64_windows-fastbuild/bin/kotlinx-html-js.js.map
  | Reason: (error: 2): The system cannot find the file specified.
```
Thanks to https://github.com/bazelbuild/rules_kotlin/blob/master/kotlin/internal/js/importer.py

Dependencies are the worst. I want to kill this python dependency but it's probably not a job I should sign up for. I could mark this example as not working on bazelci-windows. Any other ideas?
(fyi @git-str)

**Comment by alexeagle** [comment]
I'm going to just exclude it on bazelci-windows so it can merge. Fixing transitive deps isn't my windmill for today :face_with_head_bandage:

**Pull request merged**